src/System.Private.CoreLib/shared/System/SpanHelpers.cs

```diff
@@ -411,5 +412,18 @@ public static unsafe void ClearWithReferences(ref IntPtr ip, nuint pointerSizeLe
             // Write only element.
             ip = default;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int TrailingZeroCountFallback(int matches)
+        {
+            // https://graphics.stanford.edu/~seander/bithacks.html#ZerosOnRightMultLookup
+            return TrailingCountMultiplyDeBruijn[(int)(((uint)((matches & -matches) * 0x077CB531U)) >> 27)];
+        }
+
+        private static ReadOnlySpan<byte> TrailingCountMultiplyDeBruijn => new byte[32]
```

Comment:
Could it be `Lazy<ReadOnlySpan<byte>>` to avoid `byte[]` allocation on each access?

Reply:
This is taking advantage of a compiler optimization available in Roslyn and does not allocate: https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGA3AWACgUBmLbANiwCYMAVOWDAbyI3YwAcEBLANwjxyVAEpwIAEwDyAOwA2ATwDKnCDIA8wBfAB89BBB5yeMgOYBhAPYBXGTACy1uTB6dFAETgAhBNZ4ArGQwAXj0ZOAB3DC14AG0SagBdNg5WQg4MjDQAGgxsXOoADgKCgE5c7AAWAuqMElySHIxqagKm7ABWAq68gHZc2uKUzLr85v68+uap6jHscryKXInqJbzWvOKMNewxnuwm0uGMAF8CQhOgA===

Reply:
Is a magic C# optimization for `ReadOnlySpan<byte>` and `new byte[]` specifically that it directly wraps the data section of the dll instead (bit like string constants) https://github.com/dotnet/roslyn/pull/24621

Reply:
I am curious about this optimization. Would there be any perf benefit in moving the dereference out of the body, in other words:
```csharp
// In class
private static ref byte _src => ref MemoryMarshal.GetReference(TrailingCountMultiplyDeBruijn);

…
// In method body
return Unsafe.AddByteOffset(ref _src, ((uint)((matches & -matches) * 0x077CB531U)) >> 27);
```